### PR TITLE
Removing findDOMNode usage from Pivot

### LINKED ISSUE #14322: Pivot (next): remove findDOMNode usage
The new Pivot control in `react-tabs` uses [`ReactDOM.findDOMNode`](https://reactjs.org/docs/react-dom.html#finddomnode) to find the overflow button. `findDOMNode` is [deprecated in strict mode](https://reactjs.org/docs/strict-mode.html#warning-about-deprecated-finddomnode-usage) and does not work with function components, meaning this call will stop working once the component is updated to use the function component version of Button.

### PR DESCRIPTION
**Pull request by czearing**
#### Pull request checklist
- [X] Fixes #14322
- [X] Include a change request file using `$ yarn change`

#### Description of changes
1. Adding an elementRef to overflow button in Pivot
2. Removing setOverflowMenuButtonRef
3. Passing overflowMenuButtonRef as an elmentRef to `CommandButton`

#### Focus areas to test
`Pivot`

### CODE REVIEW

**Review by ecraig12345 — APPROVED** [review]

**Comment by ecraig12345** [comment]
It looks like this might actually have caused an issue, based on this screener result:
image

**Review by behowell — CHANGES_REQUESTED** [review]
It looks like overflow is broken by this change, according to the screener result that Elizabeth pasted. Nothing in this PR immediately stands out as wrong; but perhaps Button's elementRef property isn't working as expected?

**Comment by czearing** [comment]
Here's what was done to fix `Pivot's` overflow behavior after removing findDOMNode:
1.	The `BaseButton’s` ref was expected to go to the root element but it was being unnecessarily wrapped by a span. This was removed and replaced with a fragment.
2.	In Pivot.styles the star in the selector was removed as the overflow button had its visibility set to hidden after `BaseButton's` span was removed.

File: packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx:356]: This seems pretty risky to do in existing widely-used code. @khmakoto or @dzearing (if you weren't the one to suggest this) what do you think?

Comment (khmakoto) [reply packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx:356]: Yeah, I'd say we shouldn't change it in case someone is targeting that specific element in the DOM.

Comment (khmakoto) [reply packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx:356]: Ok, spoke to David offline and the reason for the span wrapper's existence is because this was written before React fragments existed and causes pretty messy bugs since the DOM structure is not inline with normal button. This is indeed a breaking change but it's fundamentally wrong to keep it as is.

My proposal would be to make the change but hold off on merging this PR until we are in the beta phase so that it merges into the compat layer in v8. And since it's going into v8 vs v7 there is less risk in introducing a breaking change.

Comment (dzearing) [reply packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx:356]: yup let's go with that, thanks all!

Comment (ecraig12345) [reply packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx:356]: The fragment change should be unblocked now that 7 has forked off, just be sure to add it in the release notes.

**Pull request closed without merging**
